src/psyclone/dynamo0p3.py

```diff
@@ -6554,42 +7515,11 @@ def basis(self, function_space):
                                  Function Space.
         :raises GenerationError: if a quadrature other than XYoZ is required.
         :raises InternalError: if the evaluator shape is not recognised.
-        '''
-        from psyclone.f2pygen import DeclGen
-        # the size of the first dimension for a
-        # basis array depends on the
-        # function space. The values are
-        # w0=1, w1=3, w2=3, w3=1, wtheta=1, w2h=3, w2v=3
-        first_dim = None
-        if function_space.orig_name.lower() in \
-           ["w0", "w3", "wtheta"]:
-            first_dim = "1"
-        elif (function_space.orig_name.lower() in
-              ["w1", "w2", "w2h", "w2v", "any_w2"]):
-            first_dim = "3"
-        else:
-            raise GenerationError(
-                "Unsupported space for basis function, "
-                "expecting one of {0} but found "
-                "'{1}'".format(VALID_FUNCTION_SPACES,
-                               function_space.orig_name))
-        ndf_name = get_fs_ndf_name(function_space)
 
+        '''
```

Comment:
This function doesn't seem to raise any generation exceptions but the docstring says it does.

Reply:
I've lost track of where this is exactly but I've corrected a couple of docstrings in the vicinity so hopefully that was it.